middleware/grpc_cluster.go

```diff
@@ -69,15 +70,15 @@ func handleClusterValidationError(err error, method string, invalidClusterValida
 // If an empty cluster label or nil logger are provided, ClusterUnaryServerInterceptor panics.
 // If the softValidation parameter is true, errors related to the cluster label validation are logged, but not returned.
 // Otherwise, an error is returned.
-func ClusterUnaryServerInterceptor(cluster string, softValidation bool, logger log.Logger) grpc.UnaryServerInterceptor {
+func ClusterUnaryServerInterceptor(cluster string, softValidation bool, invalidClusterValidations *prometheus.CounterVec, logger log.Logger) grpc.UnaryServerInterceptor {
```

Comment:
nit: it's a bit awkward to call this "invalid ... validations". Maybe just `invalidCluster`?

Reply:
Now that you say it, you've got a point. I didn't really name it myself, just copied from a corresponding client side metric in Mimir.

Reply:
I'll rename it as above, since it will match the metric name pretty well.

Reply:
Makes sense, thanks!